app/controllers/account_subscriptions_controller.rb

```diff
@@ -74,4 +76,16 @@ def reauthenticate_user
       ),
     )["auth_uri"]
   end
+
+  def has_subscription?(subscriber)
+    subscriptions = GdsApi.email_alert_api.get_subscriptions(id: subscriber.fetch("id")).to_h.fetch("subscriptions", [])
```

Comment:
We're now making two calls to this endpoint for users who aren't already subscribed